WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/POSCardPresentPaymentMessageView.swift

```diff
@@ -0,0 +1,54 @@
+import SwiftUI
+
+struct POSCardPresentPaymentMessageViewModel {
+    let showProgress: Bool?
+    let title: String?
+    let message: String?
+    let buttons: [CardPresentPaymentsModalButtonViewModel]
+
+    init(showProgress: Bool? = nil, title: String? = nil, message: String? = nil, buttons: [CardPresentPaymentsModalButtonViewModel] = []) {
+        self.showProgress = showProgress
+        self.title = title
+        self.message = message
+        self.buttons = buttons
+    }
+}
+
+struct POSCardPresentPaymentMessageView: View {
+    let viewModel: POSCardPresentPaymentMessageViewModel
+
+    var body: some View {
+        VStack {
+            Rectangle()
+                .frame(width: 100, height: 100)
+                .foregroundColor(Color(UIColor.wooCommercePurple(.shade80)))
+            if let showProgress = viewModel.showProgress, showProgress == true {
+                ProgressView()
+                    .progressViewStyle(IndefiniteCircularProgressViewStyle(
+                        size: Layout.progressViewSize,
+                        lineWidth: Layout.progressViewLineWidth))
+            }
+            if let title = viewModel.title {
+                Text(title)
+                    .font(Font.system(size: 48, weight: .bold))
+                    .foregroundColor(Color(UIColor.wooCommercePurple(.shade80)))
+            }
+            if let message = viewModel.message {
+                Text(message)
+                    .font(Font.system(size: 20))
```

Comment:
nit: for the font, is it possible to map to the semantic style like `body`?

Reply:
I could, but I do not know what are "rule" is, to either follow strictly Figma font sizes or use the semantic ones?

Reply:
If it's not specified in the design, I used to map it manually with the typography guide pzlDWpqXNhwq8kzOGMmMEh/Woo-Mobile---iOS-Component-Library?node-id=2249-3664&t=gbYHZGnwJwDEaM3S-0 (if the link is hard to work with, it's also linked in P91TBi-3PM-p2 > Component Libraries. But with the custom font that's used in the design, the mapping might not work. Just a non-blocking nit suggestion.